Source/Core/VideoBackends/Metal/MTLUtil.mm

```diff
@@ -81,6 +82,32 @@
 
 void Metal::Util::PopulateBackendInfoFeatures(VideoConfig* config, id<MTLDevice> device)
 {
+  // Initialize DriverDetails first so we can use it later
+  DriverDetails::Vendor vendor = DriverDetails::VENDOR_UNKNOWN;
+  if ([[device name] containsString:@"NVIDIA"])
+    vendor = DriverDetails::VENDOR_NVIDIA;
+  else if ([[device name] containsString:@"AMD"])
+    vendor = DriverDetails::VENDOR_ATI;
+  else if ([[device name] containsString:@"Intel"])
+    vendor = DriverDetails::VENDOR_INTEL;
+  else if ([[device name] containsString:@"Apple"])
+    vendor = DriverDetails::VENDOR_APPLE;
+  double version = 0;
+  if (@available(macOS 13, iOS 16, *))
+    version = 3.0;
```

Comment:
Perhaps it would be better to use macOS version instead of the Metal version for DriverDetails, since Apple may fix driver bugs in minor macOS updates.

Reply:
Then wouldn't we have to figure out some versioning system that unifies iOS and macOS versions?  Then again if they did fix a bug in a minor update, there's no guarantee it would come to iOS and macOS at the same time...

Reply:
Technically, you don't need to worry about iOS since Dolphin has no official iOS support. (That being said, having more things upstreamed would make my fork easier to maintain...)

Reply:
OK, so I had a chat with @JMC47 and they said it should be fine to merge some iOS support upstream.

I think we should use the OS version instead of the Metal version for DriverDetails, and just have separate entries for iOS and macOS. That's probably the easiest and cleanest way.

Reply:
Okay
I put it in as `major * 100 + minor`, but if you prefer we can enjoy our floating point rounding fun and go for `major + minor / 100`